js/client-battle.js

```diff
@@ -421,8 +421,20 @@
 			}
 			return '<button name="openTimer" class="button timerbutton' + timerTicking + '"><i class="fa fa-hourglass-start"></i> ' + time + '</button>';
 		},
+		timerInterval: 0,
+		timerTick: function () {
+			if (typeof this.battle.kickingInactive == "number") this.battle.kickingInactive--;
+			this.updateTimer();
+		},
 		updateTimer: function () {
-			this.$('.timerbutton').replaceWith(this.getTimerHTML());
+			var timerButton = this.$('.timerbutton');
+			if (timerButton.length) {
+				timerButton.replaceWith(this.getTimerHTML());
+				if (!this.timerInterval) this.timerInterval = setInterval(this.timerTick.bind(this), 1000);
```

Comment:
Are you sure you want to be setting this in `updateTimer`? You should most likely be doing this in `getTimerHTML` itself. `updateTimer` is only called for specific updates.